tests/unit/library/test_json.py

```diff
@@ -2023,7 +2041,122 @@ def test_ignored_keys_allowed_as_properties():
         "type": "object",
         "properties": {
             key: {"type": "string"} for key in IGNORED_KEYS
-        }
+        },
+        "required": list(IGNORED_KEYS),
     }
     target_obj = {key: "value" for key in IGNORED_KEYS}
     generate_and_check(target_obj, schema_obj)
+
+class TestRequiredProperties:
+    schema_obj = {
+        "type": "object",
+        "properties": {
+            "a": {"type": "string"},
+            "b": {"type": "number"},
+            "c": {"type": "boolean"},
+        },
+        "additionalProperties": True
+    }
+    ALL_REQUIRED = ["a", "b", "c"]
+    SOME_REQUIRED_SUBSETS = [[], ["a"], ["b"], ["c"], ["a", "b"], ["a", "c"], ["b", "c"], ["a", "b", "c"]]
```

Comment:
Minor thing, but this looks like an empty list counts as 'some' ?

Reply:
Fair point. Looks like the "some" positive tests make the other positive tests redundant, yeah?

Reply:
I'd agree that `SOME_REQUIRED_SUBSETS` looks a lot like the full powerset, so probably does cover everything. I don't feel strongly, but it might be worth having the 'all' and 'none' cases as separate tests, to make those most basic cases easier to find. As I said, not a very strong feeling.

Reply:
I think that explicit is better than implicit, and these tests are pretty quick. I'll leave things as they are. Thanks!